aerleon/aclcheck_cmdline.py

```diff
@@ -17,72 +17,104 @@
+        formatter_class=RawTextHelpFormatter,
+    )
+    _parser.add_argument(
         '--definitions-directory',
-        dest='definitions',
+        dest='definitions_directory',
         help='definitions directory',
-        default='./def',
     )
-    _parser.add_option(
+    _parser.add_argument(
         '--base-directory',
-        dest='basedir',
+        dest='base_directory',
         help='The base directory to look for include files.',
-        default='',
     )
-    _parser.add_option(
+    _parser.add_argument(
         '-p',
         '--policy-file',
         dest='pol',
         help='policy file',
-        default='./policies/pol/sample_paloalto_yaml.yaml',
+        required=True,
+    )
+    _parser.add_argument(
+        '--config-file',
+        dest='config_file',
+        help='config file',
     )
-    _parser.add_option(
-        '-d', '--destination', dest='dst', help='destination IP', default='200.1.1.1'
+    _parser.add_argument('-d', '--destination', dest='dst', help='destination IP')
+    _parser.add_argument(
+        '-s',
+        '--source',
+        dest='src',
+        help='source IP',
     )
-    _parser.add_option('-s', '--source', dest='src', help='source IP', default='any')
-    _parser.add_option(
+    _parser.add_argument(
         '--proto',
         '--protocol',
         dest='proto',
         help='Protocol (tcp, udp, icmp, etc.)',
-        default='tcp',
-    )
-    _parser.add_option(
-        '--dport', '--destination-port', dest='dport', help='destination port', default='80'
     )
-    _parser.add_option(
-        '--sport', '--source-port', dest='sport', help='source port', default='1025'
-    )
-    (FLAGS, unused_args) = _parser.parse_args()
+    _parser.add_argument('--dport', '--destination-port', dest='dport', help='destination port')
+    _parser.add_argument('--sport', '--source-port', dest='sport', help='source port')
+    FLAGS = _parser.parse_args()
+
+    default_flags = {
+        'base_directory': './policies',
+        'definitions_directory': './def',
+        'pol': None,
+        'config_file': None,
+        'dst': '200.1.1.1',
+        'src': 'any',
+        'proto': 'any',
+        'dport': '80',
+        'sport': '1025',
+    }
 
-    defs = naming.Naming(FLAGS.definitions)
-    with open(FLAGS.pol) as f:
+    configs = {}
+    configs.update(default_flags)
+
+    if not (FLAGS.base_directory and FLAGS.definitions_directory):
+        common_flags = frozenset(['base_directory', 'definitions_directory'])
+        common_configs = config.load_config(config_file=FLAGS.config_file)
+        common_configs = {
+            key: value for key, value in common_configs.items() if key in common_flags
+        }
+        configs.update(common_configs)
+
+    # Grab common configs from aerleon.yml
+    configs.update({flag: value for flag, value in vars(FLAGS).items() if value})
+
+    defs = naming.Naming(configs['definitions_directory'])
+
+    with open(configs['pol']) as f:
```

Comment:
This can raise an error when then file does not exist and should be caught providing a better error message.